notification/notification-job/src/main/scala/org/sunbird/job/notification/function/NotificationFunction.scala

```diff
@@ -80,55 +88,145 @@ class NotificationFunction(config: NotificationConfig,  @transient var notificat
+                } else {
+                    logger.info("Notification sent failure")
+                    handleFailureMessage(event, context, metrics)
+                }
                 //}
             }
             else logger.info("NotificationService:processMessage action name is incorrect: " + actionValue + "for message id:" + event.msgId)
         }
         else logger.info("NotificationService:processMessage event data map is either null or empty for message id:" + event.msgId)
     }
-    
+
     protected def getMaxIterations: Int = {
         maxIterations = config.max_iteration_count_samza_job
         if (maxIterations == 0) maxIterations = MAXITERTIONCOUNT
         maxIterations
     }
-    
+
     def sendEmailNotification(notificationMap: scala.collection.immutable.HashMap[String, AnyRef]) = {
         import scala.collection.JavaConverters._
-        logger.info("NotificationService:sendEmailNotification map: "+ notificationMap)
-        val emailIds : util.List[String] = notificationMap.get(IDS).get.asInstanceOf[List[String]].asJava
-        logger.info("NotificationService:sendEmailNotification emailids: "+ emailIds)
-        val templateMap : util.Map[String, AnyRef] = notificationMap.get(TEMPLATE).get.asInstanceOf[scala.collection.immutable.Map[String, AnyRef]].asJava
+        logger.info("NotificationService:sendEmailNotification map: " + notificationMap)
+        val emailIds: util.List[String] = notificationMap.get(IDS).get.asInstanceOf[List[String]].asJava
+        logger.info("NotificationService:sendEmailNotification emailids: " + emailIds)
+        val templateMap: util.Map[String, AnyRef] = notificationMap.get(TEMPLATE).get.asInstanceOf[scala.collection.immutable.Map[String, AnyRef]].asJava
+        val template = new util.HashMap[String, Any]()
+        templateMap.forEach((i, j) => {
+            template.put(i, j)
+        })
         val config = notificationMap.get(CONFIG).get.asInstanceOf[scala.collection.immutable.Map[String, AnyRef]].asJava
         val subject = config.get(SUBJECT).asInstanceOf[String]
-        val emailText = templateMap.get(DATA).asInstanceOf[String]
+        val pdata = new util.HashMap[String, Any]()
+        pdata.put(VER, "1.0")
+        pdata.put(ID, "org.sunbird.platform")
+        val context = new util.HashMap[String, Any]()
+        context.put(PDATA, pdata)
+        var message = new String()
+        if (notificationMap.get(TEMPLATE) != null && templateMap.get(DATA) != null) {
+            message = getDataMessage(templateMap.get(DATA).toString, templateMap.get(PARAMS).asInstanceOf[util.Map[String, Any]], context)
+            template.put(DATA, message)
+        } else if (templateMap.get(ID) != null && notificationMap.get(TEMPLATE) != null) {
+            val dataString = createNotificationBody(templateMap, context)
+            template.put(DATA, dataString)
+        }
+        val emailText = template.get(DATA).asInstanceOf[String]
```

Comment:
just check templateMap.get(DATA) != null -- to handle sunbird cases... in else part check ID is not null and use the readVm method... 
do check emailText is not empty before creating EmailRequest --- if emailText is empty return false.